File: pinyougou_cart_web/src/main/webapp/js/service/cartService.js
app.service("cartService", function ($http) {
    /**
     * 查询购物车列表
     * @returns {*}
     */
    this.findCartList = function () {
        return $http.get('cart/findCartList.do')
    }
    /**
     * 添加购物车
     * @param num
     * @returns {*}
     */
    this.addGoodsToCartList = function (itemId, num) {
        return $http.post('cart/saveCartList.do?itemId=' + itemId + '&num=' + num)
    }

    /**
     * 计算总价钱
     * @param cartList
     */
    this.sum = function (cartList) {
        var totalValue = {totalNum: 0, totalMoney: 0}
        for (var i = 0; i < cartList.length; i++) {
            var cart = cartList[i];
            for (var j = 0; j < cart.orderItemList.length; j++) {
                var orderItem = cart.orderItemList[j]
                totalValue.totalNum += orderItem.num;
                totalValue.totalMoney += orderItem.totalFee;
            }
        }
        return totalValue;
    }

    //获取地址列表
    this.findAddressList = function () {
        return $http.get('address/findListByLoginUser.do');
    }

    //增加
    this.add = function (order) {
        return $http.post('../order/add.do', order);
    }

})

File: pinyougou_cart_interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojo.Cart;

import java.util.List;

/**
 * 购物车服务接口
 */
public interface CartService {

    /**
     * 添加购物车
     *
     * @param cartList 购物车列表
     * @param itemId   用户ID
     * @param num      商品的数量
     * @return
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

    /**
     * 查询购物车
     *
     * @param username
     * @return
     */
    public List<Cart> findCartListFromRedis(String username);

    /**
     * 存储购物车列表
     *
     * @param username
     * @param cartList
     */
    public void saveCartListToRedis(String username, List<Cart> cartList);


    /**
     * 合并购物车
     *
     * @param cartList1
     * @param cartList2
     * @return
     */
    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);


}
